doc/tutorials/using_cat_on_up2.rst

```diff
@@ -0,0 +1,157 @@
+`cache-mask-vm`:
+
+.. figure:: images/using_cat_up2.png
+   :align: center
+   :name: cache-mask-vm
+
+   Cache usage with Cache Mask
+
+
+CAT Support on ACRN
+*******************
+
+As described at ``Intel® 64 and IA-32 Architectures Software Developer’s Manual``,
+chapter 17.17, volume 3. There are 3 steps to use CAT: detecting the CAT capability;
+setup cache mask array MSRs, which is refered as Class of Service(CLOS) array; select
+one of the CLOS array for the CPU, that will be the cache mask of the CPU.
+ACRN integrated the usage of CAT into vm configuration. The CLOS array must be defined
+at the board config source code, and the VM need to specify which CLOS to use in its vm
+config data. If the platform has CAT, the CLOS array will be write to CLOS MSRs at cpu
+init time, and the VM will set CLOS for VCPU at VM launch time.
+The details is shown in :numref:`acrn-cat-hld`:
+
+.. figure:: images/acrn_cat_hld.png
+   :align: center
+   :name: acrn-cat-hld
+
+   CAT Support on ACRN
+
+Tuning CAT at HV debug shell
+############################
+
+When at ACRN HV debug shell, user can use 'cpuid', 'wrmsr/rdmsr' debug commands to
+enumarate CAT capability and tune CAT parameters. We will use UP2 board as an example,
+the UP2 can use its serial port for HV shell, how to setup UP2 serial port please refer
+to `Getting started guide for UP2 board <https://projectacrn.github.io/latest/getting-started/up2.html>`
+
+#. Check CAT ability with cupid, first run 'cpuid 0x10 0x0', the return value ebx[bit 2]
+   report the L2 CAT is supported. Then run 'cpuid 0x10 0x2' to query L2 CAT capability,
+   the return eax[bit 4:0] report cache mask has 8 bit, edx[bit 15:0] report 4 CLOS are
+   supported
+
+   .. code-block:: none
+
+     ACRN:\>cpuid 0x10 0x0
+     cpuid leaf: 0x10, subleaf: 0x0, 0x0:0x4:0x0:0x0
+
+     ACRN:\>cpuid 0x10 0x2
+     cpuid leaf: 0x10, subleaf: 0x2, 0x7:0x0:0x0:0x3
+
+#. Check pcpu ids of each VM, the 'vcpu_list' shows that VM0 is running on PCPU0,
+   and VM1 is running on PCPU1
+
+   .. code-block:: none
+
+     ACRN:\>vcpu_list
+
+     VM ID    PCPU ID    VCPU ID    VCPU ROLE    VCPU STATE
+     =====    =======    =======    =========    ==========
+       0         0          0        PRIMARY       Running
+       1         1          0        PRIMARY       Running
+
+#. Set CLOS with wrmsr <reg_num> <value>, we want VM1 to use the lower 6 ways of cache,
+   so CLOS0 is set to 0xc0 for the upper 2 ways, and CLOS1 is set to 0x3f for the lower
+   6 ways.
+
+   .. code-block:: none
+
+     ACRN:\>wrmsr  -p1 0xd10  0xc0
+     ACRN:\>wrmsr  -p1 0xd11  0x3f
+
+#. Attach COS1 to PCPU1, because MSR IA32_PQR_ASSOC [bit 63:32], so write 0x100000000
+   to it ti use CLOS1
+
+   .. code-block:: none
+
+     ACRN:\>wrmsr   –p1   0xc8f    0x100000000
+
+Configure CAT for VM with VM Configuration
+##########################################
+
+#. CAT on ACRN can be enabled and configured by modifying source code, so the first
```

Comment:
Is this the only way to enable CAT on ACRN?  (No Kconfig option to set?)  Modifying the ACRN source code to enable options doesn't seem to be a sustainable approach to maintain customer configuration information.

Reply:
Hi, Kinder, we will update this DOC as soon as Kconfig can be used for CAT feature

Reply:
OK, should we wait for that or merge the document as it is and submit a Kconfig fix later?

Reply:
As I understand it, modifying the source code is the only option for now. If true, we should merge this as it is today. We can update the doc once a Kconfig-based mechanism exists for this.